lib/libc/picolibc/Kconfig

```diff
@@ -15,9 +19,14 @@ config PICOLIBC_USE_MODULE
 
 # force TLS when using toolchain with TLS support
 config PICOLIBC_USE_TOOLCHAIN
-	bool
-	default y if !PICOLIBC_USE_MODULE
+	bool "Picolibc from toolchain"
+	depends on "$(TOOLCHAIN_HAS_PICOLIBC)" = "y"
 	select THREAD_LOCAL_STORAGE if ARCH_HAS_THREAD_LOCAL_STORAGE && TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
+	help
+	  Use picolibc included with the toolchain.
+	  This is required when using the C++ library.
```

Comment:
This should probably use the same language as `REQUIRES_FULL_LIBCPP` because there's also the minimal C++ library which can be used with the picolibc module (or minimal libc, I assume).
```
This is required when using a full C++ standard library.
```
Should we reference the config variable here?

Reply:
Updated the help message to mention the Kconfig as well, I guess it's fine for the help section to be a little bit more verbose?